### PR TITLE
Update some dependencies

### PR DESCRIPTION
**Pull request by golinski**
Update many dependencies to newest (or at least newer) versions.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -41,13 +41,13 @@
         <starter.min.bundles.count>126</starter.min.bundles.count>
 
         <!-- versions to be replaced in the feature files -->
-        <asm.version>9.1</asm.version>
-        <jackrabbit.version>2.20.2</jackrabbit.version>
+        <asm.version>9.2</asm.version>
+        <jackrabbit.version>2.21.8</jackrabbit.version>
```
Comment (rombert) [inline pom.xml:45]: I think we should keep using the 2.20.x stream for Jackrabbit. 2.21.x is unstable and will be released as 2.22.x

**Review by rombert — CHANGES_REQUESTED** [review]
Thanks for the comprehensive update @golinski ! This LGTM, but please leave out the Jackrabbit updates.

File: pom.xml (hunk 1 above)
Comment (golinski) [reply pom.xml]: Thanks for the clarification. Upgraded to 2.20.4 instead.

**Comment by rombert** [comment]
Thanks for the contribution @golinski !

**Pull request merged**